src/SignalR/clients/csharp/Client/test/UnitTests/HubConnectionTests.cs

```diff
@@ -568,6 +563,40 @@ public async Task UploadStreamErrorSendsStreamComplete()
         }
     }
 
+    [Fact]
+    [LogLevel(LogLevel.Trace)]
+    public async Task ActiveUploadStreamWhenConnectionClosesObservesException()
+    {
+        using (StartVerifiableLog())
+        {
+            var connection = new TestConnection();
+            var hubConnection = CreateHubConnection(connection, loggerFactory: LoggerFactory);
+            await hubConnection.StartAsync().DefaultTimeout();
+
+            var channel = Channel.CreateUnbounded<int>();
+            var invokeTask = hubConnection.InvokeAsync<object>("UploadMethod", channel.Reader);
+
+            var invokeMessage = await connection.ReadSentJsonAsync().DefaultTimeout();
+            Assert.Equal(HubProtocolConstants.InvocationMessageType, invokeMessage["type"]);
+
+            // Not sure how to test for unobserved task exceptions, best I could come up with is to check that we log where there once was an unobserved task exception
```

Comment:
This is what I did to test for them in the gRPC client: https://github.com/grpc/grpc-dotnet/blob/87680d504ad72340162111f3eaeb6285292bb773/test/Grpc.Net.Client.Tests/AsyncUnaryCallTests.cs#L254-L262

Reply:
Are you sure that works? I tried using `TaskScheduler.UnobservedTaskException` but it never triggered.